src/benchmark/file_io_read_micro_benchmark.hpp

```diff
@@ -2,48 +2,68 @@
+      const auto start = std::chrono::high_resolution_clock::now();
+      if (verbose)
+        std::cout << "Creating random_indexes for: " << size_parameter << std::endl;
+      random_indexes = generate_random_indexes(number_of_elements);
+      const auto stop = std::chrono::high_resolution_clock::now();
+      const auto elapsed = std::chrono::duration_cast<std::chrono::seconds>(stop - start);
+      if (verbose)
+        std::cout << "Elapsed seconds: " << elapsed.count() << std::endl;
+      last_size_parameter = size_parameter;
+    }
+  }
+
   void SetUp(::benchmark::State& state) override {
     const auto size_parameter = state.range(0);
+
     NUMBER_OF_BYTES = _align_to_pagesize(size_parameter);
     NUMBER_OF_ELEMENTS = NUMBER_OF_BYTES / uint32_t_size;
-    filename = "benchmark_data_" + std::to_string(size_parameter) + ".txt";
 
-    auto fd = int32_t{};
-    if (!std::filesystem::exists(filename)) {
-      numbers = generate_random_positive_numbers(NUMBER_OF_ELEMENTS);
-      control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
-      Assert(((fd = creat(filename.c_str(), O_WRONLY)) >= 1),
-             close_file_and_return_error_message(fd, "Create error: ", errno));
-      chmod(filename.c_str(), S_IRWXU);  // enables owner to rwx file
-      Assert((write(fd, std::data(numbers), NUMBER_OF_BYTES) == NUMBER_OF_BYTES),
-             close_file_and_return_error_message(fd, "Write error: ", errno));
-    } else {
-      Assert(((fd = open(filename.c_str(), O_RDONLY)) >= 0),
-             close_file_and_return_error_message(fd, "Open error: ", errno));
-      const auto* map = reinterpret_cast<uint32_t*>(mmap(NULL, NUMBER_OF_BYTES, PROT_READ, MAP_PRIVATE, fd, 0));
-      Assert((map != MAP_FAILED), close_file_and_return_error_message(fd, "Mapping Failed: ", errno));
-      const auto map_span_view = std::span{map, NUMBER_OF_ELEMENTS};
-      control_sum = std::accumulate(map_span_view.begin(), map_span_view.end(), uint64_t{0});
-    }
-    close(fd);
+    numbers = generate_random_positive_numbers(NUMBER_OF_ELEMENTS);
+    control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
+
+    filename = "benchmark_data_" + std::to_string(size_parameter) + ".bin";
+    std::ofstream file(filename, std::ios::binary);
+    file.write(reinterpret_cast<const char*>(numbers.data()), numbers.size() * sizeof(uint32_t));
+    chmod(filename.c_str(), S_IRWXU);  // enables owner to rwx file
+    file.close();
+  }
+
+  void TearDown(::benchmark::State& /*state*/) override {
+    Assert(std::remove(filename.c_str()) == 0, "Remove error: " + std::strerror(errno));
   }
 
  protected:
   const ssize_t uint32_t_size = ssize_t{sizeof(uint32_t)};
+  // MAX_NUMBER_OF_ELEMENTS = max. read/write according to man page / uint32_t_size
+  // = 2,147,479,552 bytes / 4 bytes = 536,869,888
+  const uint64_t MAX_NUMBER_OF_ELEMENTS = uint64_t{536'869'888};
```

Comment:
Why do we use uint64_t here? And isn't the `MAX_NUMBER_OF_ELEMENTS` the 2 Billion from above?
I guess it is divided by 4 here, since we did a lot of this in recent Storage Manager Stuff, but there we incremented a pointer, which then advances the number of Bits of the pointers type. 

Here, the variable is only used to tell the upper boundary of read() / write(). I guess we should use `std::numeric_limits<int32_t>::max();` instead.

Actually, we could also put this as `constexpr` somewhere.

Reply:
Yes, it is the 2 billion above but in elements not in bytes. I don't know with which we want to go, but thought that comparing elements is more intuitive.